src/Symfony/Component/Console/Descriptor/Descriptor.php

```diff
@@ -70,6 +70,22 @@ protected function write($content, $decorated = false)
         $this->output->write($content, false, $decorated ? OutputInterface::OUTPUT_NORMAL : OutputInterface::OUTPUT_RAW);
     }
 
+    /**
+     * Sorts alphabetically given array of objects using ->getName(), and returns it.
+     *
+     * @param array $objects
+     *
+     * @return array
+     */
+    protected function sortByName(array $objects): array
+    {
+        \uasort($objects, function ($objectA, $objectB) {
+            return \strcmp($objectA->getName(), $objectB->getName());
```

Comment:
Is there a type/interface with `->getName()` method? It's very weak typing ot anything that can cause "calling method on non-object/invalid-object error"

Reply:
@TomasVotruba The initial thought was to sort `Symfony\Component\Console\Input\InputArgument` and `Symfony\Component\Console\Input\InputOption`, unfortunately no they don't implement/extend a common interface/class. Now, after reading the documentation, I've realized that my idea of sorting `InputArgument` wasn't good because they are ordered so rendering them in a different order than the one you're supposed to pass them would lead users to extremely hard situations to understand/debug... That's why I think we should apply the sorting only on `InputOption`. 

That said, I can rename the function `sortByName` to something clearer about its purpose like:
- `sortInputOption`
- `sortOption`

And then explicitly type hint the callback arguments as `InputOption $optionA, InputOption $optionB`.

Also after reading what @nicolas-grekas said on the issue #28108 if the idea of having `$priority` property on `InputOption` is accepted, this function could become `orderOptionsByPriority()`. But because it's specific to `InputOption` should it seats in the abstract `Descriptor` still?

Reply:
I see, in that case it should be at least:

```diff
   /**
     * Sorts alphabetically given array of objects using ->getName(), and returns it.
     *
-    * @param InputArgument[]|InputOption[] $objects
+    *
     *
     * @return array
     */
    protected function sortByName(array $objects): array
```